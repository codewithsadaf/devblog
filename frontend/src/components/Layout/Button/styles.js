//imports
import styled, { css } from "styled-components";

const handleButtonVariant = {
    default: (theme) => css`
        background: ${theme.colors.primaryColor};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.primaryColor};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            background: transparent;
            color: ${theme.colors.primaryColor};
            border: 2px solid ${theme.colors.primaryColor};
            transition: ${theme.transition.tran_03};
        }
    `,
    outline: (theme) => css`
        background: transparent;
        color: ${theme.colors.primaryColor};
        border: 2px solid ${theme.colors.primaryColor};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            background: ${theme.colors.primaryColor};
            color: ${theme.colors.white};
            border: 2px solid ${theme.colors.primaryColor};
            transition: ${theme.transition.tran_03};
        }
    `,

    success: (theme) => css`
        background: ${theme.colors.success};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.success};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            opacity: 0.8;
            transition: ${theme.transition.tran_03};
        }
    `,

    danger: (theme) => css`
        background: ${theme.colors.danger};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.danger};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            opacity: 0.8;
            transition: ${theme.transition.tran_03};
        }
    `,
};

const handleUppercase = () => css`
    text-transform: uppercase;
`;

const handleMaxWidth = () => css`
    width: 100%;
    padding: 1em 0;
`;

const handleSmallStyles = (theme) => css`
    padding: 7px 30px;
    border-radius: 4px;
    font-size: ${theme.font.sizes.small};
    border: 1px solid;

    &:hover {
        border: 1px solid;
    }
`;

export const ButtonContainer = styled.button`
    ${({ theme, buttonVariant, uppercase, maxWidth, isSmall }) => css`
        padding: 0.5em 1.8em;
        font-size: 1em;
        border: none;
        font-weight: bold;
        cursor: pointer;
        ${handleButtonVariant[buttonVariant](theme)};
        ${uppercase && handleUppercase()};
        ${maxWidth && handleMaxWidth()};
        ${isSmall && handleSmallStyles(theme)};
    `}
`;
