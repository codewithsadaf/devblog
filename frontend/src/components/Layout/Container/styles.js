//imports
import styled from "styled-components";

export const Container = styled.div`
    min-height: 60vh;
    margin-bottom: 5em;
    padding: 0 2em;
`;
