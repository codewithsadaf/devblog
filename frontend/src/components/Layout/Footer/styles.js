//imports
import styled, { css } from "styled-components";

export const FooterWrapper = styled.footer`
    ${({ theme }) => css`
        height: 250px;
        width: 100%;
        display: flex;
        align-items: center;
        justify-content: center;
        flex-direction: column;
        background-color: ${theme.colors.footerBackground};
        text-align: center;
    `}
`;
export const FooterHeading = styled.h3`
    ${({ theme }) => css`
        @media ${theme.media.lteMedium} {
            font-size: ${theme.font.sizes.medium};
            line-height: ${theme.font.line_height.h6};
        }
    `}
`;

export const CopyrightText = styled.span`
    display: block;
    margin-top: 0.8em;
`;
