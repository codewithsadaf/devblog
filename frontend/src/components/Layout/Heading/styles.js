//imports
import styled, { css } from "styled-components";

const headingSize = {
    huge: (theme) => css`
        font-size: ${theme.font.sizes.xhuge};
        line-height: ${theme.font.line_height.h1};
        ${mediaFont(theme)};
    `,
    big: (theme) => css`
        font-size: ${theme.font.sizes.huge};
        line-height: ${theme.font.line_height.h2};
    `,
    large: (theme) => css`
        font-size: ${theme.font.sizes.xxlarge};
        line-height: ${theme.font.line_height.h3};
    `,
    medium: (theme) => css`
        font-size: ${theme.font.sizes.medium};
        line-height: ${theme.font.line_height.h6};
    `,
    small: (theme) => css`
        font-size: ${theme.font.sizes.small};
        line-height: ${theme.font.line_height.small};
    `,
    xsmall: (theme) => css`
        font-size: ${theme.font.sizes.xsmall};
        line-height: ${theme.font.line_height.small};
    `,
};

const mediaFont = (theme) => css`
    @media ${theme.media.lteMedium} {
        font-size: ${theme.font.sizes.xlarge};
        line-height: ${theme.font.line_height.h4};
    }
`;

export const Container = styled.h1`
    ${({ theme, size, uppercase, bold }) => css`
        text-transform: ${uppercase ? "uppercase" : "none"};
        font-weight: ${bold ? "bold" : 500};
        ${headingSize[size](theme)};
    `}
`;
