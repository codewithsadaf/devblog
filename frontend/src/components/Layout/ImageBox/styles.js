//imports
import styled, { css } from "styled-components";

const handleBigStyle = () => css`
    width: 100%;
    height: auto;
`;

export const Container = styled.div`
    ${({ theme }) => css``}
`;

export const ImageElm = styled.img`
    ${({ rounded }) => css`
        width: 150px;
        height: 150px;
        object-fit: cover;
        display: block;
        margin: 1em auto;
        border-radius: ${rounded ? "50%" : "7px"};
    `}
`;
