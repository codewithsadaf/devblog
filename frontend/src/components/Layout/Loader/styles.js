//imports
import styled, { css } from "styled-components";

export const Container = styled.div`
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100vh;
    background: rgba(0, 0, 0, 0.6);
    z-index: 99999;
    pointer-events: all;
`;

export const Circle = styled.div`
    ${({ theme }) => css`
        &,
        &::before,
        &::after {
            border-radius: 50%;
            width: 2.5em;
            height: 2.5em;
            animation-fill-mode: both;
            animation: bblFadInOut 1.8s infinite ease-in-out;
        }

        & {
            color: ${theme.colors.white};
            font-size: 7px;
            position: relative;
            text-indent: -9999em;
            transform: translateZ(0);
            animation-delay: -0.16s;
        }

        &::before,
        &:after {
            content: "";
            position: absolute;
            top: 0;
        }

        &::before {
            left: -3.5em;
            animation-delay: -0.32s;
        }

        &::after {
            left: 3.5em;
        }

        @keyframes bblFadInOut {
            0%,
            80%,
            100% {
                box-shadow: 0 2.5em 0 -1.3em;
            }
            40% {
                box-shadow: 0 2.5em 0 0;
            }
        }
    `}
`;
