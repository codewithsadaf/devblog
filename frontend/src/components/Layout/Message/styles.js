//imports
import styled, { css } from "styled-components";
//react icons
import { IoClose } from "react-icons/io5";

export const Container = styled.div`
    ${({ theme }) => css`
        display: flex;
        align-items: center;
        justify-content: space-between;
        position: fixed;
        bottom: 30px;
        left: 50%;
        transform: translateX(-50%);
        padding: 25px 30px;
        min-width: 25%;
        border-radius: 7px;
        z-index: 9999;
        font-weight: bold;
        box-shadow: rgba(0, 0, 0, 0.24) 0px 3px 8px;
        transition: ${theme.transition.tran_03};

        &.success {
            background-color: ${theme.colors.success};
            color: ${theme.colors.white};
        }

        &.error {
            background-color: ${theme.colors.danger};
            color: ${theme.colors.white};
        }

        @media ${theme.media.lteMedium} {
            bottom: 0;
            right: 50%;
            left: 50%;
            min-width: 90%;
            transform: translate(-50%, -50%);
        }
    `}
`;

export const CloseIcon = styled(IoClose)`
    ${({ theme }) => css`
        color: ${theme.colors.white};
        cursor: pointer;
        font-size: 19px;
        margin-left: 0.3em;
    `}
`;

export const MessageText = styled.span`
    display: block;
`;
