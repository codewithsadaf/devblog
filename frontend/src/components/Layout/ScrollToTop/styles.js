//imports
import styled, { css } from "styled-components";
//react icons
import { FaChevronUp } from "react-icons/fa";

const handleActive = () => css`
    visibility: visible;
    opacity: 1;
    pointer-events: all;
`;

export const Container = styled.div`
    ${({ theme, isActive }) => css`
        position: fixed;
        width: 50px;
        height: 50px;
        background: ${theme.colors.primaryColor};
        bottom: 30px;
        right: 30px;
        text-align: center;
        cursor: pointer;
        line-height: 50px;
        color: ${theme.colors.white};
        font-size: ${theme.font.sizes.large};
        border-radius: 7px;
        transition: ${theme.transition.tran_05};
        border: 2px solid transparent;
        visibility: hidden;
        opacity: 0;
        pointer-events: none;
        transition: 0.5s;
        z-index: 999;
        ${isActive && handleActive()};

        &:hover {
            transform: scale(1.05);
            transition: all 0.5s;
            background: transparent;
            color: ${theme.colors.primaryColor};
            border: 2px solid ${theme.colors.primaryColor};
            transition: ${theme.transition.tran_05};
        }

        @media ${theme.media.lteMedium} {
            visibility: hidden;
            opacity: 0;
            display: none;
            pointer-events: none;
        }
    `}
`;

export const ArrowTop = styled(FaChevronUp)`
    ${({ theme }) => css``}
`;
