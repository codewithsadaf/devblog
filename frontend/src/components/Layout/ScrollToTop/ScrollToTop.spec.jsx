//imports
import { fireEvent, screen, waitFor } from "@testing-library/react";
import { act } from "react-dom/test-utils";
//styles
import { renderTheme } from "../../../styles/render-theme";
import { theme } from "../../../styles/theme";
//components
import { ScrollToTop } from "./";

//window.scrollTo mock
const spyScrollTo = jest.fn();
Object.defineProperty(global.window, "scrollTo", { value: spyScrollTo });

describe("<ScrollToTop />", () => {
    describe("Rendering", () => {
        beforeEach(() => {
            renderTheme(
                <div style={{ height: "1000px" }}>
                    <ScrollToTop />
                </div>
            );
        });

        it("should render ScrollToTop", () => {
            const scrollToTopIcon = screen.getByLabelText("Scroll to top Icon");

            expect(scrollToTopIcon).toBeInTheDocument();
            expect(scrollToTopIcon.parentElement).toBeInTheDocument();
        });

        it("should render ScrollToTop when user windowY is above 100px", () => {
            const scrollToTopIcon = screen.getByLabelText("Scroll to top Icon");

            act(() => {
                fireEvent.scroll(window, {
                    target: { scrollY: 250 },
                });
            });
            waitFor(() => {
                expect(scrollToTopIcon).toHaveStyleRule(
                    "visibility",
                    "visible"
                );
            });
        });

        it("should not show ScrollToTop when user windowY is less then 100px", () => {
            const scrollToTopIcon =
                screen.queryByLabelText("Scroll to top Icon");

            act(() => {
                fireEvent.scroll(window, {
                    target: { scrollY: 50 },
                });
            });
            waitFor(() => {
                expect(scrollToTopIcon).not.toHaveStyleRule(
                    "visibility",
                    "visible"
                );
            });
        });

        it("should not show ScrollToTop on mobile", () => {
            const scrollToTopIcon =
                screen.queryByLabelText("Scroll to top Icon");

            expect(scrollToTopIcon.parentElement).toHaveStyleRule(
                "display",
                "none",
                { media: theme.media.lteMedium }
            );
        });
    });

    describe("Functionality", () => {
        it("should scroll user scroll/window to top on ScrollToTop click", () => {
            renderTheme(
                <div style={{ height: "1000px" }}>
                    <ScrollToTop />
                </div>
            );

            const scrollToTopIcon =
                screen.queryByLabelText("Scroll to top Icon");

            act(() => {
                fireEvent.scroll(window, {
                    target: { scrollY: 100 },
                });
            });
            act(() => {
                fireEvent.click(scrollToTopIcon);
            });

            waitFor(() => {
                expect(spyScrollTo).toHaveBeenCalledWith({
                    top: 0,
                    behavior: "smooth",
                });
            });
        });
    });

    describe("Snapshot", () => {
        it("should match snapshot", () => {
            const { container } = renderTheme(
                <div style={{ height: "1000px" }}>
                    <ScrollToTop />
                </div>
            );

            expect(container.firstChild).toMatchSnapshot();
        });
    });
});
