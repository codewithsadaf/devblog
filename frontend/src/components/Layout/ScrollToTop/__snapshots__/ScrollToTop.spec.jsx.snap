// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScrollToTop /> Snapshot should match snapshot 1`] = `
.c0 {
  position: fixed;
  width: 50px;
  height: 50px;
  background: #000;
  bottom: 30px;
  right: 30px;
  text-align: center;
  cursor: pointer;
  line-height: 50px;
  color: #fff;
  font-size: 1.4em;
  border-radius: 7px;
  -webkit-transition: all 500ms ease-in-out;
  transition: all 500ms ease-in-out;
  border: 2px solid transparent;
  visibility: hidden;
  opacity: 0;
  pointer-events: none;
  -webkit-transition: 0.5s;
  transition: 0.5s;
  z-index: 999;
}

.c0:hover {
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
  background: transparent;
  color: #000;
  border: 2px solid #000;
  -webkit-transition: all 500ms ease-in-out;
  transition: all 500ms ease-in-out;
}

@media (max-width:768px) {
  .c0 {
    visibility: hidden;
    opacity: 0;
    display: none;
    pointer-events: none;
  }
}

<div
  style="height: 1000px;"
>
  <div
    class="c0"
  >
    <svg
      aria-label="Scroll to top Icon"
      class=""
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
      />
    </svg>
  </div>
</div>
`;
